homeassistant/components/smarty/fan.py

```diff
@@ -61,76 +59,64 @@ def supported_features(self):
         """Return the list of supported features."""
         return SUPPORT_SET_SPEED
 
-    @property
-    def speed_list(self):
-        """List of available fan modes."""
-        return SPEED_LIST
-
     @property
     def is_on(self):
         """Return state of the fan."""
-        return self._state
+        return bool(self._smarty_fan_speed)
+
+    @property
+    def speed_count(self) -> Optional[int]:
```

Comment:
```py
def speed_count(self) -> int:
```

Reply:
Fixed in https://github.com/home-assistant/core/pull/46941